website/blog/2019-06-04-hello-reason-react-native.md

```diff
@@ -0,0 +1,129 @@
+mind, but at a cost. The cost of being harder to start with for new comers to
+ReasonML.
+
+While some of us were thinking about some simplifications, some others, involved
+in Reason React incoming changes, directly started a new approach for this
+bindings, following
+[the zero-cost approach offered by reason-react 0.7.0](https://reasonml.github.io/reason-react/blog/2019/04/10/react-hooks).
+
+[CCA](https://github.com/cca-io) started this effort on what is now know as the
+new `reason-react-native` bindings, that remove most of the additional
+JavaScript code produced by the bindings.
+
+## Zero-cost
+
+We mean almost zero additional cost. So the code you produce with the bindings
+will most of the time directly hit React Native JavaScript without an additional
+cost on top of it.
+
+Here are some random example
+
+- [`bs-react-native` `View`](https://github.com/reasonml-community/bs-react-native/blob/958cd4e3a5ffa303304a0b2404cd53b5f49e649f/bs-react-native/src/components/ViewProps.bs.js)
+  vs
+  [`reason-react-native` `View`](https://github.com/reasonml-community/bs-react-native/blob/958cd4e3a5ffa303304a0b2404cd53b5f49e649f/reason-react-native/src/components/View.bs.js)
+- [`bs-react-native` `Style`](https://github.com/reasonml-community/bs-react-native/blob/958cd4e3a5ffa303304a0b2404cd53b5f49e649f/bs-react-native/src/style.bs.js)
+  vs
+  [`reason-react-native` `Style`](https://github.com/reasonml-community/bs-react-native/blob/958cd4e3a5ffa303304a0b2404cd53b5f49e649f/reason-react-native/src/apis/Style.bs.js)
+- [`bs-react-native` `Platform`](https://github.com/reasonml-community/bs-react-native/blob/958cd4e3a5ffa303304a0b2404cd53b5f49e649f/bs-react-native/src/platform.bs.js)
+  vs
+  [`reason-react-native` `Platform`](https://github.com/reasonml-community/bs-react-native/blob/958cd4e3a5ffa303304a0b2404cd53b5f49e649f/reason-react-native/src/apis/Platform.bs.js)
+
+Some modules are even empty or with some minor `require()` that can be themself
+almost empty & easily removed by some JavaScript bundlers.
+
+How cool is that?
+
+## But why`reason-react-native` ?
+
+`bs-react-native` already had multiples breaking changes in the past & we didn't
+really wanted to offer a poor experience by creating a new release with tons of
+changes that won't be easy to codemod (or would require much more work from us).
+
+On top of that, `reason-react` latest release already added an important
+breaking change with
+[JSX 3](https://reasonml.github.io/reason-react/docs/en/jsx).
+
+That's why we had this idea about offering a easy & smooth transition by
+offering 2 packages to say goodbye to `bs-react-native` without pain.
+
+- `bs-react-native-jsx3-compat`
+- `reason-react-native`
+
+## Wait, what is `bs-react-native-jsx3-compat` ?
+
+`bs-react-native` is using JSX 2 so in order to use it with JSX 3 we would have
+to create bindings for the bindings to offer both JSX 2 & 3 bindings. And since
```

Comment:
The "bindings for the bindings" part is hard to understand.

Reply:
I switched to "to create an additional layer on top of the current bindings"